lib/parser.js

```diff
@@ -47,32 +49,37 @@ Parser.prototype._preprocess = function (element, context, config) {
     let isUrl = utils.isUrl(element.attribs.src);
     let includeSrc = url.parse(element.attribs.src);
     let includeSrcPath = includeSrc.path;
-    let filePath = isUrl ? element.attribs.src : path.resolve(path.dirname(context.cwf), includeSrcPath);
+    let targetFilePath = isUrl ? element.attribs.src : path.resolve(path.dirname(context.cwf), includeSrcPath);
+    let actualFilePath = targetFilePath;
+    let boilerplateFilePath = calculateBoilerplateFilePath(includeSrcPath);
+    if (!fs.existsSync(targetFilePath) && fs.existsSync(boilerplateFilePath)) {
+      actualFilePath = boilerplateFilePath;
+    }
     element.name = isInline ? 'span' : 'div';
-    element.attribs[ATTRIB_INCLUDE_PATH] = filePath;
+    element.attribs[ATTRIB_INCLUDE_PATH] = targetFilePath;
 
     if (isDynamic) {
       element.name = 'dynamic-panel';
-      element.attribs.src = filePath;
+      element.attribs.src = targetFilePath;
       if (includeSrc.hash) {
         element.attribs.fragment = includeSrc.hash.substring(1);
       }
 
       element.attribs.header = element.attribs.name || '';
       delete element.attribs.dynamic;
-      this.dynamicIncludeSrc.push({from: context.cwf, to: element.attribs.src});
+      this.dynamicIncludeSrc.push({from: context.cwf, target: element.attribs.src, actualFilePath: actualFilePath});
       return element;
     }
 
     if (isUrl) {
       return element; // only keep url path for dynamic
     }
 
-    let isIncludeSrcMd = utils.getExtName(filePath) === 'md';
+    let isIncludeSrcMd = utils.getExtName(targetFilePath) === 'md';
 
     try {
-      self._fileCache[filePath] = self._fileCache[filePath] ?
-        self._fileCache[filePath] : fs.readFileSync(filePath, 'utf8');
+      self._fileCache[actualFilePath] = self._fileCache[actualFilePath] ?
+        self._fileCache[actualFilePath] : fs.readFileSync(actualFilePath, 'utf8');
     } catch (e) {
       // Read file fail
       let missingReferenceErrorMessage = `Missing reference in: ${element.attribs[ATTRIB_CWF]}`;
```

Comment:
How about:

```js
try {
  self._fileCache[targetFilePath] = self._fileCache[targetFilePath] || fs.readFileSync(targetFilePath, 'utf8');
} catch (e) {
  // Read file fail: targetFilePath
  let boilerplateFilePath = calculateBoilerplateFilePath(includeSrcPath);
  
  try {
    self._fileCache[targetFilePath] = fs.readFileSync(boilerplateFilePath, 'utf8');
  } catch (e) {
    // Read file fail: boilerplateFilePath
    let missingReferenceErrorMessage = `Missing reference in: ${element.attribs[ATTRIB_CWF]}`;
      e.message += `\n${missingReferenceErrorMessage}`;
      this._onError(e);
      element.children = cheerio.parseHTML(utils.createErrorElement(e), true);
      return element;
  }
}
```

Reply:
Should this still be changed?

I think it should be
`self._fileCache[boilerplateFilePath] = fs.readFileSync(boilerplateFilePath, 'utf8');`
and the error reported should be the first one

Reply:
Hmm, since we're using the same pattern for `actualFilePath` in `includeFile`, let's not change this.